GitUp/Application/WelcomeWindow.m

```diff
@@ -97,33 +97,125 @@ - (void)concludeDragOperation:(id<NSDraggingInfo>)sender {
 }
 @end
 
-@interface WelcomeWindow ()
+@implementation WelcomeWindow
+
+#pragma mark - Setup
+- (void)setup {
+  self.opaque = NO;
+  self.backgroundColor = [NSColor clearColor];
+  self.movableByWindowBackground = YES;
+}
+
+- (void)awakeFromNib {
+  [self setup];
+}
+
+#pragma mark - Window
+- (BOOL)canBecomeKeyWindow {
+  return YES;
+}
+
+@end
+
+@interface WelcomeWindowControllerModel ()
+@property (assign, nonatomic, readwrite) BOOL shouldShow;
+@end
+
+@implementation WelcomeWindowControllerModel
+#pragma mark - Show
+- (void)setShouldShow {
+  self.shouldShow = YES;
+}
+- (void)setShouldHide {
+  self.shouldShow = NO;
+}
+@end
```

Comment:
I think we can just keep the old mechanism in place of explicitly telling the window (now window controller) to show or close the window.